docs/vsphere.md

```diff
@@ -1,23 +1,267 @@
+      "WaitForIP": false,
+      "Name": null
+    }
+    ```
+
+4. Create a VM from the `OVA`: 
+
+    ```
+    govc import.ova -options=options.json $(basename "${OVA_URL}")
+    ```
+
+### Create template VM from qcow2 
+
+Prerequisites:
+
+* vSphere (tested on version 6.7)
+* GOVC (tested on version 0.22.1)
+* qemu-img (tested on version 4.2.0)
+* curl or wget
+
+Procedure:
+
+1. Download the guest image in qcow2 format end export an environment variable
+   whith the name of the file.
+
+    ```
+    # The URL below is just an example
+    image_url="https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2"
+    image_name="$(basename -- "${image_url}" | sed 's/.qcow2$//g')"
+    curl -sL "${image_url}" -O .
+    ```
+
+2. Convert it to vmdk e.g.
+
+    ```
+    qemu-img convert -O vmdk -o subformat=streamOptimized "./${image_name}.qcow2" "${image_name}.vmdk"
+    ```
+
+3. Upload to vSphere using WebUI or GOVC:
+
+    Make sure to replace the parameters on the command below with the correct
+    values specific to yout vSphere environment.
+
+    ```
+    govc import.vmdk -dc=dc-1 -pool=/dc-1/host/cl-1/Resources -ds=ds-1 "./${image_name}.vmdk"
+    ```
+
+4. Inflate the created disk (see vmware [documentation][inflate_thin_virtual_disks] for more details)
+
+    ```
+    govc datastore.disk.inflate -dc dc-1 -ds ds-1 "${image_name}/${image_name}.vmdk"
+    ```
+
+5. Create a new virtual machine using that image with vSphere WebUI.
+6. During the `Customize Hardware` step:
+    1. Remove the disk present by default
+    2. Click on `ADD NEW DEVICE`, select `Existing Hard Disk` and select the
+       disk previously created.
+7. The vm is ready to be used by the `MachineController` by referencing its name in the field `.spec.template.spec.providerSpec.value.cloudProviderSpec.templateVMName` of the `MachineDeployment`.
+
+### OS images
+
+Information about supported OS versions can be found [here](./operating-system.md#supported-os-versions).
+
+#### Ubuntu
+
+Ubuntu OVA template can be foud at <https://cloud-images.ubuntu.com/releases/18.04/release/ubuntu-18.04-server-cloudimg-amd64.ova>.
+
+Follow [OVA](#create-template-vm-from-ova) template VM creation guide.
+
+#### CoreOS
+
+CoreOS OVA template can be foud at <https://stable.release.core-os.net/amd64-usr/current/coreos_production_vmware_ova.ova>.
+
+Follow [OVA](#create-template-vm-from-ova) template VM creation guide.
+
+#### RHEL
+
+Red Hat Enterprise Linux 8.x KVM Guest Image can be found at [Red Hat Customer Portal][rh_portal_rhel8].
+
+Follow [qcow2](#create-template-vm-from-qcow2) template VM creation guide.
+
+#### CentOS
+
+CentOS 7 image can be found at the following link: <https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2r>.
+
+Follow [qcow2](#create-template-vm-from-qcow2) template VM creation guide.
+
+## Provider configuration
+
+VSphere provider accepts the following configuration parameters:
+
+```yaml
```

Comment:
* backticks
* syntax hint
* no indent

Reply:
no indent because we are not inside a numbered list